daemon/start.go

```diff
@@ -11,79 +10,95 @@ import (
 	"google.golang.org/grpc"
 
 	"github.com/Sirupsen/logrus"
-	apierrors "github.com/docker/docker/api/errors"
 	"github.com/docker/docker/api/types"
 	containertypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/container"
+	"github.com/pkg/errors"
 )
 
 // ContainerStart starts a container.
 func (daemon *Daemon) ContainerStart(name string, hostConfig *containertypes.HostConfig, checkpoint string, checkpointDir string) error {
 	if checkpoint != "" && !daemon.HasExperimental() {
-		return apierrors.NewBadRequestError(fmt.Errorf("checkpoint is only supported in experimental mode"))
+		return validationError{fmt.Errorf("checkpoint is only supported in experimental mode")}
 	}
 
 	container, err := daemon.GetContainer(name)
 	if err != nil {
 		return err
 	}
 
-	if container.IsPaused() {
-		return fmt.Errorf("Cannot start a paused container, try unpause instead.")
-	}
+	err = func() error {
+		container.Lock()
```

Comment:
This would have to be the real container object.
Indeed it was a bug.
This started off as, "let me check all the states here instead of some here and some in the next call in the stack... ok need to lock... ok this should be a defer, too many early returns... oh crap this stuff is mutating the container..."